include/navigation/image_database.h

```diff
@@ -510,57 +438,63 @@ class ImageDatabase
     bool hasMetadata() const;
 
     template<class Func>
-    size_t forEachImage(const Func &func, bool greyscale = true) const
+    void forEachImage(const Func &func, size_t frameSkip = 1,
+                        bool greyscale = true) const
     {
+        BOB_ASSERT(frameSkip > 0);
+
         if (empty()) {
-            return 0;
+            return;
         }
 
         // If database consists of individual image files...
-        if (m_VideoFileName.empty()) {
-            size_t errors = 0;
-
-            /*
-             * You can't have uncaught exceptions eminating from OpenMP blocks,
-             * so we just count the number of errors thrown instead.
-             */
-            #pragma omp parallel for shared(errors)
-            for (size_t i = 0; i < m_Entries.size(); i++) {
+        if (m_VideoFilePath.empty()) {
+            std::exception_ptr eptr;
+
+            #pragma omp parallel shared(eptr)
+            #pragma omp for
+            for (size_t i = 0; i < m_Entries.size() / frameSkip; i++) {
+                /*
+                 * You can't have uncaught exceptions eminating from OpenMP blocks,
+                 * so we catch them and rethrow later.
+                 */
                 try {
-                    func(i, m_Entries[i].load(greyscale));
-                } catch (std::exception &e) {
-                    LOGE << "Error occurred in forEachImage: " << e.what();
-
-                    #pragma omp atomic
-                    errors++;
+                    func(i, m_Entries[i * frameSkip].load(greyscale));
+                } catch (...) {
+                    eptr = std::current_exception();
```

Comment:
woah! I never knew you could do this 😨

Reply:
Neat huh? It's handy for punting exceptions between threads (like we do with ``BackgroundExceptionCatcher``).